internal/logger/telemetry_store.go

```diff
@@ -40,10 +41,22 @@ type ActionEvalData struct {
 	Result string `json:"result"`
 }
 
+// RuleType is a struct describing a rule type for telemetry purposes
+type RuleType struct {
+	Name string     `json:"name"`
+	ID   *uuid.UUID `json:"id"`
+}
+
+// Profile is a struct describing a Profile for telemetry purposes
+type Profile struct {
+	Name string     `json:"name"`
+	ID   *uuid.UUID `json:"id"`
+}
+
 // RuleEvalData reports
 type RuleEvalData struct {
-	RuleName    string `json:"rule_name"`
-	ProfileName string `json:"profile_name"`
+	RuleType *RuleType `json:"ruletype"`
+	Profile  *Profile  `json:"profile"`
```

Comment:
One more note: I don't know if it's easier to have these as pointers (caller may need to allocate structure vs set individual fields) or as embedded objects, but I wouldn't worry about memory overhead costs at this point.  If you use them as structures, I think you can compare with the default to see if they are empty:

```golang
if red.RuleType == RuleType{} {
    ....
}
```

Reply:
My motivation was mostly about the memory overhead of that.

Reply:
I'd be cautious about optimizing here -- I _think_ you're trading off memory allocations vs additional objects that the GC has to track and collect.  I don't know what the right knobs are, but I'd tend to write this in ways that are easy to read and avoid errors until we have an indication that this is a performance problem (and where the problem is, because we're about to assemble a whole bunch of JSON...).